### PR TITLE
Moving to ubuntu-22.04 to resolve TSAN warnings

### PR DESCRIPTION
**Pull request by woody-apple**
https://github.com/project-chip/connectedhomeip/issues/14710 shows that TSAN errors are resolved in ubuntu-22, so moving our CI this way.

### CODE REVIEW

**Comment by woody-apple** [comment]
Tagging @andy31415

**Review by andy31415 — APPROVED** [review]

File: .github/workflows/bloat_check.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     pull_request_update:
         name: Report on pull requests
 
-        runs-on: ubuntu-latest
```
Comment (andy31415) [inline .github/workflows/bloat_check.yaml:30]: This seems ok, but does it work? The line 33 below uses chip-build which is still `FROM ubunut:focal` so I am guessing we just change host not the underlying OS.

Can we somehow validate if this is what we want or if we instead want to change our chip-build and underlying docker images?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: .github/workflows/bloat_check.yaml (hunk 1 above)
Comment (woody-apple) [reply .github/workflows/bloat_check.yaml:30]: 100%. I'll update images too. Let me update those as well

**Review by andy31415 — CHANGES_REQUESTED** [review]
Request changes based on request from @Arkq - tsan issues are likely due to glib not compiled with tsan, so changing versions for tsan may not work.

We would need some custom glib compilation for this, Arkadiusz is testing.

File: .github/workflows/examples-linux-imx.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
         name: Linux i.MX Build
         timeout-minutes: 70
 
-        runs-on: ubuntu-latest
+        runs-on: ubuntu-22.04
```
Comment (jmeg-sfy) [inline .github/workflows/examples-linux-imx.yaml:30]: Can we have a matter-linux version macro to relieve our further OS update?

**Review by jmeg-sfy — APPROVED** [review]

**Comment by Arkq** [comment]
I think that this PR can be now closed as per GitHub latest change:
https://github.blog/changelog/2022-12-01-github-actions-larger-runners-using-ubuntu-latest-label-will-now-use-ubuntu-22-04/

**Pull request closed without merging**